apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js

```diff
@@ -28,93 +30,131 @@ const TemplateEdit = compose(
-					<Toolbar>
-						<IconButton
-							className={ classNames( 'components-icon-button components-toolbar__control', {
-								'is-active': isEditing,
-							} ) }
-							label={ __( 'Change Template Part' ) }
-							onClick={ toggleEditing }
-							icon="edit"
-						/>
-					</Toolbar>
-				</BlockControls>
-			) }
-			<div
-				className={ classNames( 'template-block', {
-					[ `align${ align }` ]: align,
-				} ) }
-			>
-				{ showPlaceholder && (
-					<Placeholder
-						icon="layout"
-						label={ __( 'Template Part' ) }
-						instructions={ __( 'Select a template part to display' ) }
-					>
-						<div className="template-block__selector">
-							<PostAutocomplete
-								initialValue={ get( template, [ 'title', 'rendered' ] ) }
-								onSelectPost={ onSelectTemplate }
-								postType="wp_template_part"
+		return (
+			<Fragment>
+				{ showToggleButton && isTemplate && (
+					<BlockControls>
+						<Toolbar>
+							<IconButton
+								className={ classNames( 'components-icon-button components-toolbar__control', {
+									'is-active': isEditing,
+								} ) }
+								label={ __( 'Change Template Part' ) }
+								onClick={ toggleEditing }
+								icon="edit"
 							/>
-							{ !! template && (
-								<a href={ `?post=${ templateId }&action=edit&fse_parent_post=${ currentPostId }` }>
-									{ sprintf( __( 'Edit "%s"' ), get( template, [ 'title', 'rendered' ], '' ) ) }
-								</a>
-							) }
-						</div>
-					</Placeholder>
+						</Toolbar>
+					</BlockControls>
 				) }
-				{ showContent && (
-					<Fragment>
-						<RawHTML className="template-block__content">
-							{ get( template, [ 'content', 'rendered' ] ) }
-						</RawHTML>
-						{ ! isTemplate && (
-							<Placeholder
-								className="template-block__overlay"
-								instructions={ __(
-									'This block is part of your site template and may appear on multiple pages.'
+				<div
+					className={ classNames( 'template-block', {
+						[ `align${ align }` ]: align,
+					} ) }
+				>
+					{ showPlaceholder && (
+						<Placeholder
+							icon="layout"
+							label={ __( 'Template Part' ) }
+							instructions={ __( 'Select a template part to display' ) }
+						>
+							<div className="template-block__selector">
+								<PostAutocomplete
+									initialValue={ get( template, [ 'title', 'rendered' ] ) }
+									onSelectPost={ onSelectTemplate }
+									postType="wp_template_part"
+								/>
+								{ !! template && (
+									<a
+										href={ `?post=${ templateId }&action=edit&fse_parent_post=${ currentPostId }` }
+									>
+										{ sprintf( __( 'Edit "%s"' ), get( template, [ 'title', 'rendered' ], '' ) ) }
+									</a>
 								) }
-							>
-								<Button
-									href={ `?post=${ templateId }&action=edit&fse_parent_post=${ currentPostId }` }
-									isDefault
+							</div>
+						</Placeholder>
+					) }
+					{ showContent && (
+						<Fragment>
+							{ parsedBlock && (
+								<BlockEdit
```

Comment:
Interesting, I wasn't able to find any docs for this one. When you have time later @rodrigoi, it'd be super helpful to add what benefits this change has and what BlockEdit does in the PR Summary.

Reply:
This is one of the extremely grey areas of Gutenberg, and this still very much a work in progress that, if working as I expect, will serve a double purpose:

* fix the render bug, because blocks cheat when setting things like colour options.
* bring us one step closer to in place editing of template parts.

I'm reverse engineering this from the `Group` block, and the `ReusableBlock`, and considering the lack of documentation, this PR description will take longer to write than the actual code 😧

Reply:
Thanks @rodrigoi I think it's enough then to update the summary on what items will be improved by the PR.

cc @getdave in case you had extra context with regards to the BlockEdit component.

Reply:
could this be part of the solution to #34279 then? - been looking at this and it is currently impossible (from what I can see) to apply the theme styling to the template parts because of the way the fse blocks are currently rendered.

Reply:
The first thing I've tried was to add the missing classes to the _Editor Styles_ but there are no extension points there.
The problem with #34279 is that Gutenberg is not a WYSIWYG editor. There are bound to be differences between what you edit and what you render. I'll expand this idea there.